src/p2p/behaviour.rs

```diff
@@ -305,6 +452,23 @@ impl<Types: IpfsTypes> Behaviour<Types> {
     pub fn bitswap(&mut self) -> &mut Bitswap {
         &mut self.bitswap
     }
+
+    pub fn bootstrap(&mut self) -> Result<SubscriptionFuture<Result<(), String>>, anyhow::Error> {
+        match self.kademlia.bootstrap() {
+            Ok(id) => Ok(self.kad_subscriptions.create_subscription(id.into(), None)),
+            Err(e) => {
+                error!("kad: can't bootstrap the node: {:?}", e);
+                Err(anyhow!("kad: can't bootstrap the node: {:?}", e))
+            }
+        }
+    }
+
+    pub fn get_closest_peers(&mut self, id: PeerId) -> SubscriptionFuture<Result<(), String>> {
+        let id = id.to_base58();
```

Comment:
feels strange that the peerid multihash would be converted to base58 ... is this correct?

Reply:
yes, it matches the ids of the peers; what I'm less sure about it the one for block keys, but this one should be fine